### DISCUSSION TITLE
Synology NAS keep popping up "Container plextraktsync stopped unexpectedly. Go to Container Manager for more information." notifications

### DISCUSSION THREAD

**Discussion by Shasoosh** [Taxel/PlexTraktSync#1722] [Support] [answered]
### Confirmation

- [X] I have read the [README.md](https://github.com/Taxel/PlexTraktSync#readme) on the project homepage
- [X] I have checked if identical issue already exists
- [X] I have tried downgrading to find version that can be used as a workaround

### The problem

I'm running the container using the docker compose shared and once sync is finished, my synology NAS is sending me a notification with "Container plextraktsync stopped unexpectedly. Go to Container Manager for more information."

There are no errors and the syncs completes successfully but it seems that container closes abruptly. 

### Expected behavior

Stop the container without getting a notification.

### Steps to reproduce the behavior

Docker compose on synology NAS will reproduce this.

### Install method

docker-compose

### Version

0.27.16

### Python Version

3.12.1

### Plex Server Version

v.1.32.8.7639-fb6452ebf

### Operating System and Version

DSM 7.2.1-69057 Update 3

**Comment by glensc** [comment]
Not reproducible. Provide logs and details from your system. How we're supposed to help if you provide no info other than just your word? Show your problem!

**Comment by Shasoosh** [comment]
Sorry, other than the warnings it seemed like everything is working as it should. 
I'm using a DS920+
DSM 7.2.1-69057 Update 3
Debug is turned on in the config.yml

compose:

```
version: '3'
services:
  plextraktsync:
    image: ghcr.io/taxel/plextraktsync
    command: sync
    container_name: plextraktsync
    restart: on-failure:2
    volumes:
      - /volume2/docker/plextraktsync/config:/app/config
    environment:
      - PUID=1026
      - PGID=100
      - TZ=Israel

```

Here's the log: (Please let me know if you need anything else.)

```
2024/01/07 22:00:52	stdout	INFO     Completed full sync in 10.9 seconds                                    
2024/01/07 22:00:52	stdout	INFO     Updated liked list in 0.0 seconds                                      
2024/01/07 22:00:52	stdout	Syncing show ratings 100% ━━━━━━━━━━━━ 41/41  [ 0:00:00 < 0:00:00 , 4,768 it/s ]
2024/01/07 22:00:51	stdout	INFO     סדרות ילדים processed in 1.4 seconds                                   
2024/01/07 22:00:51	stdout	Processing סדרות ילדים 100% ━━━━━━━━━━━━━━━━ 6/6  [ 0:00:01 < 0:00:00 , 4 it/s ]
2024/01/07 22:00:50	stdout	INFO     סדרות processed in 4.0 seconds                                         
2024/01/07 22:00:50	stdout	Processing סדרות 100% ━━━━━━━━━━━━━━━━━━━ 35/35  [ 0:00:04 < 0:00:00 , 12 it/s ]
2024/01/07 22:00:46	stdout	INFO     סרטים processed in 0.8 seconds                                         
2024/01/07 22:00:46	stdout	Processing סרטים 100% ━━━━━━━━━━━━━━━━━━ 64/64  [ 0:00:00 < 0:00:00 , 234 it/s ]
2024/01/07 22:00:45	stdout	INFO     סרטי ילדים processed in 3.1 seconds                                    
2024/01/07 22:00:45	stdout	Processing סרטי ילדים 100% ━━━━━━━━━━━ 199/199  [ 0:00:03 < 0:00:00 , 143 it/s ]
2024/01/07 22:00:41	stdout	INFO     Sync Show sections: ['סדרות', 'סדרות ילדים']                           
2024/01/07 22:00:41	stdout	INFO     Sync Movie sections: ['סרטי ילדים', 'סרטים']                           
2024/01/07 22:00:41	stdout	INFO     PlexTraktSync [0.27.15]  
```

**Comment by glensc** [comment]
unfortunately, that is still not useful. how are you invoking the command? how you obtained the logs? seems to be log of different runs? also you said you enabled debug, but not even shared debug logs.

**Comment by Shasoosh** [comment]
I'm not invoking a command, I just start the container using the Synology DSM UI or via SSH `docker-compose up -d`.
The logs are from inside the container via Portainer.
I've just realised that a log file is created with DEBUG. I'm uploaded it here: https://pastebin.com/FhUmX2MC

* I initially uploaded a log the contained a lot of warnings because of a missing tv show id. I've removed that show and run the container again. The log in the pastebin is what I get if there's nothing to sync. The "Container plextraktsync stopped unexpectedly" will show regardless if something was synced or not. (although sync will be successful and present on trakt.)

**Comment by glensc** [comment]
you can upload text files to github issue. drag, drop or select them. be sure to use supported extension like .txt
- https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files

**Comment by glensc** [comment]
where is that "Container plextraktsync stopped unexpectedly" appearing? your compose file is configured to run "sync" command. that will exit process once sync is complete.what does docker inspect say for the exited container?  run the command without `-d`, what happens then?

**Comment by simonc56** [comment]
Container with succesfull PTS sync is supposed to exit with `code 0` which means it finished successfully.
Maybe your OS doesn't read `exit code 0` as expected, but it is.

**Comment by Shasoosh** [comment]
@glensc In your first comment  you wrote "Not reproducible" so I was under the impression that you are also on Synology DSM. 
It could be an OS thing altough I had containrs that did short tasks and didn't produce that notification.

<img width="269" alt="image" src="https://github.com/Taxel/PlexTraktSync/assets/2679978/909f41fd-9d70-4f04-ab04-a65d942c71f6">

It shows up on the Synology OS, also sends an email about it. Unfortunately there's no way for me to disable these just for one container. Disabling these alerts would mean stop getting alerts about containers that stopped. 

Running the command without `-d` will put out `plextraktsync exited with code 0`

**Comment by glensc** [comment]
does `docker run  -d alpine sh -c "sleep 10"` also produce that notification? i.e. it comes if a container was up only for a short time? and without `-d` there's no notification?

**Comment by Shasoosh** [comment]
`docker run  -d alpine sh -c "sleep 10"` doesn't produce the notification.

**Comment by glensc** [comment]
or adjust "sleep 10" with approximate time the plextraktsync container ran? I don't see why would it consider unexpected close if it exited with 0.

also show `docker inspect` of the container that pop up the notification. be sure not to run with `--rm`. should be in `docker ps -a` list.

**Comment by Shasoosh** [comment]
[docker inspect.txt](https://github.com/Taxel/PlexTraktSync/files/13865110/docker.inspect.txt)

It seems that for `docker-compose run --rm plextraktsync sync` I will not get the alert
but for `docker start -i plextraktsync` (an existing container) the alert will show. 
Does that make any sense to you?

**Comment by glensc** [comment]
Nothing suspicous in inspect output. I personally don't use `docker start`, so not sure what it does. but guessing, if you omit `-i`, does it behave differently?

if you don't want to re-create container for next run, you could keep the container "up" by having it run "sleep infinity" command, and then execute plextraktsync commands with `docker exec`.

**Comment by Shasoosh** [comment]
Thanks for the help. It behaves the same without the `-i`.
I'll try the `sleep infinity` solution.

**Comment by glensc** [comment]
Don't forget to close issue and post solution for other users.

**Comment by Shasoosh** [comment, marked as answer]
Couldn't figure out this beaviour. 

`docker-compose up -d` -> No alert
`docker start plextraktsync` -> No alert
run container from the UI -> Alert
Then again
`docker start plextraktsync` -> Alert (??)

I've recreated the container and now runs it only from the scheduler using `docker start plextraktsync` - No alerts so far.
If the alerts will return I'll use `docker-compose run --rm plextraktsync sync` which never showed me the alert.

**Reply by antonywalker** [reply]
I have been experiencing the exact same issue as @Shasoosh on my Synology. 

The marked solution worked for a while, but now — for some reason — no longer does. 

The error pop-up doesn't affect the functionality of the container, yet the constant notifications are quite a nuisance.

**Comment by ricardodolnl** [comment]
Any solutions?
